omegaconf/listconfig.py

```diff
@@ -101,18 +102,31 @@ def _validate_set(self, key: Any, value: Any) -> None:
                     )
 
         target_type = self._metadata.element_type
+        if target_type in (Any, None):
+            return
+
         value_type = OmegaConf.get_type(value)
-        if is_structured_config(target_type):
-            if (
-                target_type is not None
-                and value_type is not None
-                and not issubclass(value_type, target_type)
-            ):
-                msg = (
-                    f"Invalid type assigned : {type_str(value_type)} is not a "
-                    f"subclass of {type_str(target_type)}. value: {value}"
-                )
-                raise ValidationError(msg)
+
+        def raise_validation_error(
+            value_type: Any, target_type: Any, value: Any
+        ) -> None:
+            msg = (
+                f"Invalid type assigned : {type_str(value_type)} is not a "
+                f"subclass of {type_str(target_type)}. value: {value}"
+            )
+
+            raise ValidationError(msg)
```

Comment:
Adding this function here, which is only called once - is making it even harder to understand what you are actually changing here.
If it's only called once, what is the point of extracting it into a local function?

Reply:
Looking back at it doesn't make sense. I think at one point I reused this function(while testing) but now not. I'll remove it.

Reply:
Sounds good.